woocommerce/woocommerce-android
# Feature/iap payment flow

Second part of: #7925


### Description
Provides integration with In App Purchases (IAP) to purchase the eCommerce plan when creating a new store. 

The flow works like this:

1. When user kicks off the store creation flow we check for IAP eligibility. That is done with [this PR changes](https://github.com/woocommerce/woocommerce-android/pull/7924)
2. If IAP is enabled and user is eligible the native store creation flow is triggered. If not, a dialogue with an error is shown. 
3. User provides all the store details and lands on the purchase eCommerce plan screen
4. When clicking on "Create store for $69.99/month" a new site is created and then the IAP purchase flow is triggered. 
5. After IAP is purchased successfully user is taken to the next step. In this step we wait for the new site to migrate to Atomic site and Jetpack and WooCommerce plugin to get installed.

### Testing instructions

Follow these instructions: PdfdoF-1Ej-p2 (Android section) to be able to use IAP. A Google Play US account is required for this. 

- Check that purchasing eCommerce plan creates a new site. We are still investigating a potential bug on backed side pdfdoF-20N-p2#comment-3131, so it is possible that once the site is created and the plan purchased successfully, the site doesn't have the eCommerce plan. 
- Check that any error during IAP purchase flow is tracked and hides the loading spinner from the main button

### Images/gif

https://user-images.githubusercontent.com/2663464/209305881-67ebe1cb-fef6-497e-ab80-b3eb63a14555.mp4
